modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AMQPMetaDataHandler.java

```diff
@@ -55,4 +65,49 @@
 
         return underlying;
     }
+
+
+    /**
+     * Update message metadata for AMQP; to indicate the message is a compressed one.
+     *
+     * @param buf                       buffer of the original metadata
+     * @param originalMetadata          source metadata that needs to be copied
+     * @param newCompressedMessageValue Value to indicate if the message is compressed or not
+     * @return copy of the metadata as a byte array
+     */
+    public static byte[] constructMetadata(ByteBuffer buf, StorableMessageMetaData originalMetadata,
+                                           boolean newCompressedMessageValue) {
+        ContentHeaderBody contentHeaderBody = ((MessageMetaData) originalMetadata).getContentHeaderBody();
+        int contentChunkCount = ((MessageMetaData) originalMetadata).getContentChunkCount();
+        long arrivalTime = ((MessageMetaData) originalMetadata).getArrivalTime();
+        long sessionID = ((MessageMetaData) originalMetadata).getPublisherSessionID();
+
+        //Modify message metadata, to update if the message is compressed or not
+
+        MessagePublishInfo messagePublishInfo = new CustomMessagePublishInfo(originalMetadata);
+        MessageMetaData modifiedMetaData = new MessageMetaData(messagePublishInfo, contentHeaderBody, sessionID,
+                contentChunkCount, arrivalTime, newCompressedMessageValue);
+
+        final int bodySize = 1 + modifiedMetaData.getStorableSize();
```

Comment:
Usually we write final int bodySize = modifiedMetaData.getStorableSize() + 1;
Also shall we add a comment as to why we add 1 to the storable size?